pallets/cash/src/rates.rs

```diff
@@ -44,27 +42,22 @@ impl APR {
         APR(uint_from_string_with_decimals(Self::DECIMALS, s))
     }
 
-    /// exp{r * dt} where dt is change in time in milliseconds
-    // XXX why is this an index, should it be a CashIndexDelta or something?
-    //  actually why is this even related to CASH?
-    // XXX this should return a Factor
-    pub fn compound(self, dt: Timestamp) -> Result<CashIndex, MathError> {
-        let index_scale = &BigInt::from(CashIndex::ONE.0);
-        let scaled_rate: &BigInt =
-            &(index_scale * self.0 * dt / MILLISECONDS_PER_YEAR / APR::ONE.0);
-        let t1 = index_scale * index_scale * index_scale; //     1
-        let t2 = scaled_rate * index_scale * index_scale; //     x
-        let t3 = scaled_rate * scaled_rate * index_scale / 2; // x^2 / 2
-        let t4 = scaled_rate * scaled_rate * scaled_rate / 6; // x^3 / 6
-        let unscaled = t1 + t2 + t3 + t4;
-        let scaled: BigInt = unscaled / index_scale / index_scale;
-        if let Some(raw) = scaled.to_u128() {
-            Ok(CashIndex(raw))
-        } else {
-            Err(MathError::Overflow)
-        }
+    /// Official approximation of e^(r*dt) using e^x = 1 + x + x^2/2 + x^3/6 + ...
+    pub fn compound(self, dt: Timestamp) -> Result<Factor, MathError> {
```

Comment:
I think this was subject to a lot of discussions that I didn't really participate in so maybe I am missing some context, but LGTM

Reply:
I think that was probably mostly around how to do the approximation...this is just updating it to use the math functions that were added later and return a `Factor`